nebula/ui/components/VPNControllerView.qml

```diff
@@ -588,12 +588,74 @@ Item {
         id: toggle
         objectName: "controllerToggle"
 
-        anchors.bottom: parent.bottom
-        anchors.bottomMargin: 48
-        anchors.horizontalCenterOffset: 0
-        anchors.horizontalCenter: parent.horizontalCenter
-        Accessible.ignored: connectionInfoScreenVisible
-        enabled: !connectionInfoScreenVisible
+        anchors {
+            bottom: parent.bottom
+            bottomMargin: 48
+            horizontalCenterOffset: 0
+            horizontalCenter: parent.horizontalCenter
+        }
+        enabled: !connectionInfoScreenVisible && !ipInfoPanel.isOpen
+
+        Accessible.ignored: connectionInfoScreenVisible || ipInfoPanel.isOpen
+    }
+
+    VPNIPInfoPanel {
+        id: ipInfoPanel
+        objectName: "ipInfoPanel"
+
+        opacity: ipInfoPanel.isOpen ? 1 : 0
+        visible: opacity > 0
+        z: 1
+    }
+
+    VPNIconButton {
+        id: ipInfoToggleButton
+        objectName: "ipInfoToggleButton"
+
+        //% "Close"
+        property var connectionInfoCloseText: qsTrId("vpn.connectionInfo.close")
+
+        anchors {
+            right: parent.right
+            rightMargin: VPNTheme.theme.windowMargin / 2
+            top: parent.top
+            topMargin: VPNTheme.theme.windowMargin / 2
+        }
+        accessibleName: ipInfoPanel.isOpen
+            ? connectionInfoCloseText
+            : VPNl18n.ConnectionInfoIpInfoButtonLabel
+        buttonColorScheme: VPNTheme.theme.iconButtonDarkBackground
+        enabled: visible
+        opacity: visible ? 1 : 0
+        visible: connectionInfoToggleButton.visible
+            && !connectionInfoScreen.isOpen
+            && !connectionInfoScreen.isTransitioning
+        z: 1
+        onClicked: {
+            ipInfoPanel.isOpen = !ipInfoPanel.isOpen;
+        }
+        Accessible.ignored: !visible
+
+        Image {
+            property int iconSize: ipInfoPanel.isOpen
+                ? VPNTheme.theme.iconSize
+                : VPNTheme.theme.iconSize * 1.5
+
+            anchors.centerIn: ipInfoToggleButton
+            source: ipInfoPanel.isOpen
```

Comment:
Non-blocking nit: I'm having a bit of a hard time reading- I think because we're hanging a number of values off of other values (`accessibleName: ipInfoPanel.isOpen ?...`, for instance). I know the `State {}` syntax is so verbose and  cumbersome but I think it may actually be a more legible, future-friendly approach.